ksql-engine/src/main/java/io/confluent/ksql/schema/registry/KsqlSchemaRegistryClientFactory.java

```diff
@@ -61,6 +68,11 @@ public SchemaRegistryClient create() {
       restService.setSslSocketFactory(sslContext.getSocketFactory());
     }
 
-    return new CachedSchemaRegistryClient(restService, 1000);
+    return new CachedSchemaRegistryClient(restService, 1000, schemaRegistryClientConfigs);
+  }
+
+  // Visible for testing
+  ImmutableMap<String, Object> getSchemaRegistryClientConfigs() {
```

Comment:
Not a fan of exposing stuff for testing. There is generally a better way. e.g. 

```
interface RegistryClientFactory {
   CachedSchemaRegistryClient create(RestService restService, Map<String, Object> config);
}

KsqlSchemaRegistryClientFactory(final KsqlConfig config,
                                final Supplier<RestService> serviceSupplier,
                                final SslFactory sslFactory,
                                final RegistryClientFactory registryClientFactory) {
    this.sslFactory = sslFactory;
    this.serviceSupplier = serviceSupplier;
    this.configs = ImmutableMap.copyOf(
            config.valuesWithPrefixOverride(KsqlConfig.KSQL_SCHEMA_REGISTRY_PREFIX));
    this.sslFactory.configure(schemaRegistryClientConfigs);
    this.registryClientFactory = registryClientFactory;
}

public SchemaRegistryClient create() {
   ... blah
   return registryClientFactory.create(restService, configs);
}
```

Then change `shouldPassBasicAuthCredentialsToSchemaRegistryClient` to pass in a mock `RegistryClientFactory` and verify that params passed are what is expected. 

Boom! No stuff exposed just for testing. :D